omod/src/main/java/org/openmrs/module/openconceptlab/web/rest/resources/ImportResource.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    @Override
+    public Import newDelegate() {
+        return new Import();
+    }
+
+    @Override
+    public Import save(Import delegate) {
+        Import anImport = getImportService().getLastImport();
+        if (anImport != null) {
+            getImportService().ignoreAllErrors(anImport);
+        }
+        UpdateScheduler updateScheduler = getUpdateScheduler();
+        updateScheduler.scheduleNow();
+        return getImportService().getLastImport();
+    }
+
+    @Override
+    public DelegatingResourceDescription getCreatableProperties() throws ResourceDoesNotSupportOperationException {
+        return new DelegatingResourceDescription();
+    }
+
+    @Override
+    public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+        throw new ResourceDoesNotSupportOperationException();
+    }
+
+    @Override
+    public void purge(Import delegate, RequestContext context) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException();
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        if (rep instanceof FullRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("localDateStarted");
+            description.addProperty("localDateStopped");
+            description.addProperty("oclDateStarted");
+            description.addProperty("errorMessage");
+            description.addProperty("importProgress");
+            description.addProperty("importTime");
+            description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+            description.addSelfLink();
+            return description;
+        } else if (rep instanceof DefaultRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("localDateStarted");
+            description.addProperty("localDateStopped");
+            description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+            description.addSelfLink();
+            return description;
+        } else if (rep instanceof RefRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addSelfLink();
+            return description;
+        }
+        return null;
+    }
+
+    @Override
+    protected PageableResult doGetAll(RequestContext context) throws ResponseException {
+        return new NeedsPaging<Import>(getImportService().getImportsInOrder(0, 20), context);
+    }
+
+    @PropertyGetter("importProgress")
```

Comment:
@rkorytkowski Could you please take a took a this? I'm getting error when requesting full representation. [here](http://pastebin.com/Vs1EFKKe) is the stacktrace.